### ISSUE TITLE
Crash opening legal .jpg file

### ISSUE THREAD

**Issue by DougRogers** [OpenImageIO/oiio#3300]
```
            ImageSpec config;
            config["oiio:UnassociatedAlpha"] = 1;

            std::string fileName = entry.path().string();
            auto inp             = ImageInput::open(fileName.c_str(), &config);


```

**Comment by lgritz** [comment]
I can't reproduce. 

Which version of OIIO? Of libjpeg? 

Do you have a stack trace and/or can you catch it in the debugger and find out where the crash occurs?

Can you reproduce the crash using any of the command line utilities, such as `iconvert bad.jpg out.tif` or `oiiotool --stats bad.jpg`? When it's just a couple isolated lines of C++, I worry that whatever other code is surrounding it, or the specific compile flags, could be key to reproducing the behavior.

Also, if you grab the jpeg right here from this page and save with your browser, does that image fail for you? I'm not sure that a JPEG pasted in an issue survives bit-for-bit or if it has any possible alteration or cleanup done by github.com. But if you zip the file and attach the zip to the issue, then to GitHub it's just data, not an image, and will be identical when I unzip it on my end. (I have no idea if that is what's going on here, but it's worth considering.)

**Comment by DougRogers** [comment]
Attaching as .zip

[NIK2549.zip](https://github.com/OpenImageIO/oiio/files/7923250/NIK2549.zip)
![OIIO](https://user-images.githubusercontent.com/7563851/150734912-fb8ed60a-4b7e-438a-a520-3d6ce1998fd8.jpg)

OpenJpeg 2.4.0
OpenImageIO 2.3.10.1

**Comment by DougRogers** [comment]
![NIK2549](https://user-images.githubusercontent.com/7563851/150735184-dcc7fec1-1963-4106-a888-bc254e2c8e0d.jpg)

**Comment by DougRogers** [comment]
Downloaded image .jpg also fails
[recon.zip](https://github.com/OpenImageIO/oiio/files/7923295/recon.zip)
.

**Comment by DougRogers** [comment]
![image](https://user-images.githubusercontent.com/7563851/150735795-02c3ae81-7aeb-4c74-b60c-b1663c155ce7.png)

**Comment by DougRogers** [comment]
![image](https://user-images.githubusercontent.com/7563851/150735866-01c6e06d-661c-45dd-851c-b5409b312fbd.png)

**Comment by DougRogers** [comment]
attr value in decode_xmp_node(xmp.cpp) is suspect

![image](https://user-images.githubusercontent.com/7563851/150736295-74832ee2-a3eb-4c53-8b3b-f87e950c4cd4.png)

**Comment by lgritz** [comment]
Wow, interesting, not where I expected a problem. Looking into it...

**Comment by DougRogers** [comment]
I am getting this error on a lot of files.   Different extensions as well.

**Comment by DougRogers** [comment]
The _isspace_ function accepts an int.  The str.front() routine is returning a signed char.  Converting the char to int results in a negative value.  This is checked is the isspace function and causes the assert.

```
extern "C" int __cdecl _chvalidator(int const c, int const mask)
{
    _ASSERTE(c >= -1 && c <= 255);
    return _chvalidator_l(nullptr, c, mask);
}

static __forceinline int fast_check_initial_locale(int const c, int const mask) throw()
{
    #ifdef _DEBUG
    return _chvalidator(c, mask);
    #else
    return __acrt_locale_get_ctype_array_value(__acrt_initial_locale_data._public._locale_pctype, c, mask);
    #endif
}

extern "C" int (__cdecl isspace)(int const c)
{
    return fast_check_current_locale(c, _SPACE);
}


```

**Comment by lgritz** [comment]
Wowee, so isspace() takes an int, but doesn't gracefully handle any int value? A value of -3 will *assert* instead of just saying, you know, this isn't one of the space characters? You'd think I would have seen this issue pop up some time in the last 35 years of using C. OK, fixing...

**Comment by lgritz** [comment]
Well, at least that explains why I couldn't reproduce -- the assertion only happens on Windows, and only for a debug build.
(Hmmm... I should add a Windows debug build to our CI at some point.)

**Comment by lgritz** [comment]
Fix proposed in #3310
